packages/core/database/src/migrations/internal-migrations/5.0.0-convert-identifiers-long-than-max-length.ts

```diff
@@ -0,0 +1,246 @@
+  // because even if it fails for some other reason, the schema sync will recreate them anyway
+  // Therefore, we wrap this in a nested transaction (considering we are running this migration in a transaction)
+  // so that we can suppress the error
+  try {
+    await knex.transaction(async (trx) => {
+      if (client === 'mysql' || client === 'mariadb') {
+        await trx.raw(
+          `ALTER TABLE \`${full.tableName}\` RENAME INDEX \`${full.indexName}\` TO \`${short.indexName}\``
+        );
+      } else if (client === 'pg' || client === 'postgres') {
+        await trx.raw(`ALTER INDEX "${full.indexName}" RENAME TO "${short.indexName}"`);
+      } else if (client === 'sqlite' || client === 'better') {
+        // SQLite doesn't support renaming, so we have to drop and recreate it
+        await recreateIndexSqlite(trx, full.indexName, short.indexName);
+      } else {
+        debug('No db client name matches, not creating index');
+      }
+    });
+  } catch (err) {
+    debug(`error creating index: ${JSON.stringify(err)}`);
+  }
+};
+
+// Select the definition of an index, drop it, and then recreate it
+// That way, we do not reacreate from the model definition and potentially lose user modifications to it
+const recreateIndexSqlite = async (knex: Knex, oldIndexName: string, newIndexName: string) => {
+  if (oldIndexName === newIndexName) {
+    debug(`not dropping and recreating index ${oldIndexName} because it hasn't changed`);
+    return;
+  }
+
+  try {
+    await knex.transaction(async (trx) => {
+      // Get the CREATE INDEX statement used for this index
+      const indexInfo = await trx
+        .select('sql')
+        .from('sqlite_master')
+        .where('type', 'index')
+        .andWhere('name', oldIndexName)
+        .first();
+
+      if (indexInfo && indexInfo.sql) {
+        // Attempt to precisely target the index name in the CREATE INDEX statement
+        const pattern = new RegExp(
+          `(CREATE\\s+(UNIQUE\\s+)?INDEX\\s+(IF\\s+NOT\\s+EXISTS\\s+)?)${oldIndexName}(\\s+ON)`,
+          'i'
+        );
+        const replacement = `$1${newIndexName}$4`;
+        const newIndexSql = indexInfo.sql.replace(pattern, replacement);
+
+        // Drop the existing index
+        await trx.raw(`DROP INDEX IF EXISTS ??`, [oldIndexName]);
+
+        // Recreate the index with a new name
+        await trx.raw(newIndexSql);
+      } else {
+        debug(`sqlite index ${oldIndexName} not found or could not retrieve definition.`);
+      }
+    });
+  } catch (err) {
+    debug(`error recreating sqlite index${JSON.stringify(err)}`);
+  }
+};
+
+const findDiffs = (shortMap: Metadata, options: MetadataOptions) => {
+  const diffs = {
+    tables: [],
+    columns: [],
+    indexes: [],
+  } as IdentifierDiffs;
+
+  const shortArr = Array.from(shortMap.entries());
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  shortArr.forEach(([key, shortObj], index) => {
```

Comment:
```suggestion
  shortArr.forEach(([, shortObj], index) => {
```

SInce you don't need to keys, why aren't you doing an Object.values above?

Reply:
Haven't you seen I love keeping around variables for posterity? 😆